File: lookup/lookup-service-client/src/main/java/net/es/lookup/records/ErrorRecord.java
package net.es.lookup.records;

import net.es.lookup.common.ReservedKeys;
import net.es.lookup.common.ReservedValues;
import net.es.lookup.common.exception.RecordException;

/**
 * Author: sowmya
 * Date: 4/30/13
 * Time: 6:07 PM
 */
public class ErrorRecord extends Record {

    public ErrorRecord() {

        super(ReservedValues.RECORD_VALUE_TYPE_ERROR);
    }

    public void setErrorMessage(String errorMessage) throws RecordException {
        if(errorMessage != null){
            this.add(ReservedKeys.ERROR_MESSAGE, errorMessage);
        } else {
            throw new RecordException("Error message is null");
        }

    }

    public String getErrorMessage(){
        return (String) this.getValue(ReservedKeys.ERROR_MESSAGE);
    }

    public void setErrorCode(int responseCode) {
        this.add(ReservedKeys.ERROR_CODE, responseCode);
    }

    public int getErrorCode() {
        return (Integer) this.getValue(ReservedKeys.ERROR_CODE);
    }


}

File: lookup/simple-lookup-service-client-python/src/main/python/net/es/lookup/records/ErrorRecord.py
__author__ = 'student5'
import Record
from net.es.lookup.common.exception import RecordException


class ErrorRecord(Record):

    def __init__(self):
        super('error') #TODO: un-hardcode this

    def setErrorMessage(self, errorMessage):
        if errorMessage is not None:
            self.add('error-message', errorMessage) #TODO: un-hardcode this
        else:
            raise RecordException("Error message is null")

    def getErrorMessage(self):
        return str(self.getValue('error-message'))

    def setErrorCode(self, responseCode):
        self.add('error-code', responseCode)

    def getErrorCode(self):
        return int(self.getValue('error-code'))